src/v0/destinations/klaviyo/transformV2.js

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable  array-callback-return */
+const get = require('get-value');
+const { ConfigurationError, InstrumentationError } = require('@rudderstack/integrations-lib');
+const { EventType, MappedToDestinationKey } = require('../../../constants');
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require('./config');
+const {
+  batchSubscriptionRequestV2,
+  constructProfile,
+  subscribeUserToListV2,
+  buildRequest,
+  buildSubscriptionRequest,
+  getTrackRequests,
+  fetchTransformedEvents,
+  addSubcribeFlagToTraits,
+} = require('./util');
+const {
+  constructPayload,
+  getFieldValueFromMessage,
+  removeUndefinedAndNullValues,
+  handleRtTfSingleEventError,
+  addExternalIdToTraits,
+  adduserIdFromExternalId,
+  flattenJson,
+} = require('../../util');
+
+/**
+ * Main Identify request handler func
+ * The function is used to create/update new users and also for subscribing
+ * users to the list.
+ * DOCS: 1. https://developers.klaviyo.com/en/reference/create_or_update_profile
+ *       2. https://developers.klaviyo.com/en/reference/subscribe_profiles
+ * @param {*} message
+ * @param {*} category
+ * @param {*} destination
+ * @returns one object with keys profile and subscription(conditional) and values as objects
+ */
+const identifyRequestHandler = (message, category, destination) => {
+  // If listId property is present try to subscribe/member user in list
+  const { listId } = destination.Config;
+  const payload = removeUndefinedAndNullValues(constructProfile(message, destination, true));
+  const response = { profile: payload };
+  let traitsInfo = getFieldValueFromMessage(message, 'traits');
+  const mappedToDestination = get(message, MappedToDestinationKey);
```

Comment:
We needed 4 lines of code for retl flow and rest all the things or flow is same hence kept it here.
Additionally if an event is through retl or through event stream can be known only through message body hence it has to be controlled at event level